### PR TITLE
Add RandomRotate for transforms

### PR DESCRIPTION
**Pull request by hkzhang95**

### CODE REVIEW

File: mmseg/datasets/pipelines/transforms.py [hunk 1]
```diff
@@ -463,6 +464,73 @@ def __repr__(self):
...
+               rotation_degree,
+               flags=cv2.INTER_LINEAR,
+               borderValue=0):
+        """Rotate  ``img``"""
+        h, w = img.shape[:2]
+        center = ((w - 1) * 0.5, (h - 1) * 0.5)
+        matrix = cv2.getRotationMatrix2D(center, rotation_degree, 1)
+        img = cv2.warpAffine(
+            img, matrix, (w, h), flags=flags, borderValue=borderValue)
+        return img
```
Comment (xvjiarui) [inline mmseg/datasets/pipelines/transforms.py:499]: I suggest making `mmcv.imrotate` support multiple interpolation modes.

**Comment by xvjiarui** [comment]
Hi @hkzhang95 
Thanks for your contribution. Would mind open a PR in mmcv to support interpolations in `imrotate`?

**Comment by hkzhang95** [comment]
Ok, I have made a pull request: [mmcv #545](https://github.com/open-mmlab/mmcv/pull/545).

File: mmseg/datasets/pipelines/transforms.py [hunk 2]
```diff
@@ -463,6 +463,60 @@ def __repr__(self):
...
+            results['rotation'] = rotation
+        if results['rotation']:
+            img = results['img']
+            rotation_degree = self.get_rotation_degree()
+
+            # rotate the image and semantic segmentation map
+            img = mmcv.imrotate(img, rotation_degree)
+            results['img'] = img
+            for key in results.get('seg_fields', []):
+                results[key] = self.rotate(
```
Comment (xvjiarui) [inline mmseg/datasets/pipelines/transforms.py:509]: We may switch to `mmcv.imroate` for this.

**Comment by xvjiarui** [comment]
Hi @hkzhang95 
Since you created a PR from master branch, I could not push to your master master from my side. So I created a new PR #215, which integrated your commit history. 
Sorry about that.

**Pull request closed without merging**